hazelcast/src/main/java/com/hazelcast/monitor/impl/GlobalIndexStats.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    @Override
+    public long getQueryCount() {
+        return queryCount;
+    }
+
+    @Override
+    public void incrementQueryCount() {
+        QUERY_COUNT.incrementAndGet(this);
+    }
+
+    @Override
+    public long getHitCount() {
+        return hitCount;
+    }
+
+    @Override
+    public long getTotalHitLatency() {
+        return totalHitLatency;
+    }
+
+    @Override
+    public double getTotalNormalizedHitCardinality() {
+        return (double) totalNormalizedHitCardinality / PRECISION;
+    }
+
+    @Override
+    public long getInsertCount() {
+        return insertCount;
+    }
+
+    @Override
+    public long getTotalInsertLatency() {
+        return totalInsertLatency;
+    }
+
+    @Override
+    public long getUpdateCount() {
+        return updateCount;
+    }
+
+    @Override
+    public long getTotalUpdateLatency() {
+        return totalUpdateLatency;
+    }
+
+    @Override
+    public long getRemoveCount() {
+        return removeCount;
+    }
+
+    @Override
+    public long getTotalRemoveLatency() {
+        return totalRemoveLatency;
+    }
+
+    @Override
+    public long getMemoryCost() {
+        return IndexHeapMemoryCostUtil.estimateMapCost(entryCount, ordered, queryableEntriesAreCached) + valuesMemoryCost;
+    }
+
+    @Override
+    public void onInsert(long timestamp, IndexOperationStats operationStats, Index.OperationSource operationSource) {
+        if (operationStats.getEntryCountDelta() == 0) {
+            // no entries were inserted
+            return;
+        }
+
+        if (operationSource == Index.OperationSource.User) {
+            TOTAL_INSERT_LATENCY.addAndGet(this, System.nanoTime() - timestamp);
+            INSERT_COUNT.incrementAndGet(this);
+        }
+        ENTRY_COUNT.incrementAndGet(this);
+        VALUES_MEMORY_COST.addAndGet(this, operationStats.getMemoryCostDelta());
+    }
+
+    @Override
+    public void onUpdate(long timestamp, IndexOperationStats operationStats, Index.OperationSource operationSource) {
+        if (operationSource == Index.OperationSource.User) {
+            TOTAL_UPDATE_LATENCY.addAndGet(this, System.nanoTime() - timestamp);
+            UPDATE_COUNT.incrementAndGet(this);
+        }
+        VALUES_MEMORY_COST.addAndGet(this, operationStats.getMemoryCostDelta());
+    }
+
+    @Override
+    public void onRemove(long timestamp, IndexOperationStats operationStats, Index.OperationSource operationSource) {
+        if (operationStats.getEntryCountDelta() == 0) {
+            // no entries were removed
+            return;
+        }
+
+        if (operationSource == Index.OperationSource.User) {
+            TOTAL_REMOVE_LATENCY.addAndGet(this, System.nanoTime() - timestamp);
+            REMOVE_COUNT.incrementAndGet(this);
+        }
+        ENTRY_COUNT.decrementAndGet(this);
+        VALUES_MEMORY_COST.addAndGet(this, operationStats.getMemoryCostDelta());
+    }
+
+    @Override
+    public void onClear() {
+        entryCount = 0;
```

Comment:
Since `entryCount` and `valuesMemoryCost` are managed by their respective updaters, don't we need to use the updaters here as well? I am not sure in which circumstances this `onClear` is called. I just wanted to double check your reasoning here.